### PR TITLE
Update keys for basicauth URL IDP to use OpenID standard claim names

### PR DESCRIPTION
**Pull request by liggitt**

### CODE REVIEW

**Comment by liggitt** [comment]
@brenton @detiber any concerns about updating the expected keys in the JSON returned from the basicauth identity provider to use standard OpenID claim names (http://openid.net/specs/openid-connect-core-1_0.html#StandardClaims)?

This PR changes "id" to "sub" and "login" to "preferred_username".

```
{
  "sub":"014fbff9a07c",
  "name":"Bob User",
  "email":"bob@example.com",
  "preferred_username":"bob"
}
```

**Comment by liggitt** [comment]
@deads2k quick review

**Comment by liggitt** [comment]
Updated docs at https://github.com/openshift/openshift-docs/pull/297

Updated example IDP at https://github.com/openshift/sti-basicauthurl/issues/2

File: pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go [hunk 1]
```diff
@@ -99,14 +101,21 @@ func (a *Authenticator) AuthenticatePassword(username, password string) (user.In
 		return nil, false, err
 	}
 
-	if len(remoteUserData.ID) == 0 {
+	if len(remoteUserData.Subject) == 0 {
 		return nil, false, errors.New("Could not retrieve user data")
 	}
+	identity := authapi.NewDefaultUserIdentityInfo(a.providerName, remoteUserData.Subject)
+
+	if len(remoteUserData.Name) > 0 {
```
Comment (deads2k) [inline pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go]: Not that I dislike cleanliness of data, but why bother?  We're already checking at the other end and I'm hard pressed to explain a significant distinction between present and empty and not-present in this case.

**Comment by deads2k** [comment]
lgtm.  Feel free to merge once the other guys are happy.

**Comment by brenton** [comment]
@liggitt, I'm OK with the change if we can get it in today.  I'm assuming you'll have an update for http://docs.openshift.org/latest/architecture/authentication.html#BasicAuthPasswordIdentityProvider when this merges.

When you mentioned this to me originally on irc I was thinking you were changing something with the identity provider configuration.  I don't think @detiber will be affected by this change so we can go ahead and merge it in.  I'll update the STI basic auth identity provider for beta3 to be compatible with this change.

File: pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go (hunk 1 above)
Comment (liggitt) [inline pkg/auth/authenticator/password/basicauthpassword/basicauthpassword.go]: just doing my part to keep etcd memory usage down

**Comment by liggitt** [comment]
[merge]

**Pull request merged**